src/Data/V/Linear.hs

```diff
@@ -59,40 +48,33 @@ module Data.V.Linear
   , caseNat
   ) where
 
+import Data.V.Internal.Linear.V
+import Prelude.Linear.Internal
+import qualified Unsafe.Linear as Unsafe
 import qualified Data.Functor.Internal.Linear.Functor as Data
 import qualified Data.Functor.Internal.Linear.Applicative as Data
 import qualified Data.Functor.Internal.Linear.Traversable as Data
-import Data.Kind (Type)
-import Data.Proxy
-import Data.Type.Equality
-import Data.Vector (Vector)
-import qualified Data.Vector as Vector
-import GHC.Exts (Constraint, proxy#)
 import GHC.TypeLits
-import Prelude
-  ( Eq
-  , Ord
-  , Int
-  , Bool(..)
-  , Either(..)
-  , Maybe(..)
-  , fromIntegral
-  , error
-  , (-))
-import qualified Prelude as Prelude
-import Prelude.Linear.Internal
-import qualified Unsafe.Linear as Unsafe
+import qualified Data.Vector as Vector
+
+{- Developers Note
+
+To avoid a common circular dependence, we moved the data type to
+Data.V.Internal.Linear.V and moved the instances here. The common import issue
+is as follows. Dupable depends on @V@ yet the instances of @V@ depend on
+a variety of things (data functors, control functors, traversable) which
+often end up depending on dupable. By moving the instances here, we
+can make sure that Data.Unrestricted.Internal.Dupable only depends on the data
+type defintion in Data.V.Linear.V and does not require any of the dependencies
+of the instances.
 
-newtype V (n :: Nat) (a :: Type) = V (Vector a)
-  deriving (Eq, Ord, Prelude.Functor)
-  -- Using vector rather than, say, 'Array' (or directly 'Array#') because it
-  -- offers many convenience function. Since all these unsafeCoerces probably
-  -- kill the fusion rules, it may be worth it going lower level since I
-  -- probably have to write my own fusion anyway. Therefore, starting from
-  -- Vectors at the moment.
+Remark: We tried to create Data.V.Internal.Linear.Instances but for some
+reason GHC couldn't export the instances from there.
```

Comment:
This sounds very unlikely. What did you actually tried to do, and what didn't work?

Reply:
I tried to do exactly what's done in `Data.Unrestricted` and that failed.